### PR TITLE
feat: create db bootstrap API

### PR DESCRIPTION
**Pull request by Alxandr**

### CODE REVIEW

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/CreateDatabaseRoleTask.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            CREATE ROLE "{_roleName}"
+            WITH LOGIN
+            PASSWORD '{pw.Password}'
+            """;
+
+        try
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+        catch (PostgresException ex) when (ex.SqlState == "42710")
```
Comment (andreasisnes) [inline src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/CreateDatabaseRoleTask.cs:46]: kunne dette vært en const med kommentar
```DUPLICATE OBJECT - Attempt to create an object that already exists in the database. Occurs when trying to create a duplicate object such as a table, index, or sequence.```

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/Program.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+TokenCredential cred;
+if (builder.Environment.IsDevelopment())
+{
+    cred = new DefaultAzureCredential();
+}
+else
+{
+    cred = new ManagedIdentityCredential();
```
Comment (andreasisnes) [inline src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/Program.cs:15]: I do believe that `ManagedIdentityCredential` defaults to using system assigned identity if not specified a client_id, we are using a user assigned identity, just need to inject it.

We can also make an exception for this service to use SystemAssigned as this service doesn't depends on external conn during boot

**Comment by andreasisnes** [comment]
missing a solution file?

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/BootstrapDatabasePipeline.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        };
+
+        var serverConnString = connStringBuilder.ToString();
+        await using var serverConn = new NpgsqlConnection(serverConnString);
+        await context.RunTask(
+            "Connecting to database server",
+            (_, ct) => serverConn.OpenAsync(ct),
+            cancellationToken);
+
+        var migratorUser = await context.RunTask(new CreateDatabaseRoleTask(secretClient, serverConn, $"{DatabaseName}_migrator", Resources.User), cancellationToken);
```
Comment (andreasisnes) [inline src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/BootstrapDatabasePipeline.cs:75]: There have been talks if an .NET app should not have their own DB and that API's should have their own schema and share database - just like the situation is between RR and Access mgmt. However, I do prefer the 1 one to 1 mapping as you try to generalize here.

**Review by andreasisnes — APPROVED** [review]

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/CreateDatabaseRoleTask.cs (hunk 1 above)
Comment (Alxandr) [reply src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/CreateDatabaseRoleTask.cs:46]: Ah - det er egentlig en const - har bare glemt å bruke den. Will fix.

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/BootstrapDatabasePipeline.cs (hunk 3 above)
Comment (Alxandr) [reply src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/BootstrapDatabasePipeline.cs:75]: I'm not enforcing that at all - this allows databases to be shared, but also to be re-used if that's what you want.

**Comment by Alxandr** [comment]
> missing a solution file?

Intentional. Wanna deal with solutions separately - including automation.

File: src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/Program.cs (hunk 2 above)
Comment (Alxandr) [reply src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/Program.cs:15]: Added config for client id.

**Pull request merged**